### PR TITLE
Add specs for the remaining cli options

### PR DESCRIPTION
**Pull request by AndreaCrotti**
Add remaining specs and a test to make sure validation works as expected.

### CODE REVIEW

File: test/carve/main_test.clj [hunk 1]
```diff
@@ -62,6 +64,11 @@ test-resources/app/app.clj:9:1 app/ignore-me")
     (is (thrown? clojure.lang.ExceptionInfo
                  (run-main "{:paths [src test]}"))))
 
+  (testing "Generate random options validate"
+    (doseq [o (g/sample (s/gen ::main/opts))]
+      ;; the paths needs to exist to to simplify the test we just hard code it
+      (is nil? (main/validate-opts! (assoc o :paths ["."])))))
```
Comment (AndreaCrotti) [inline test/carve/main_test.clj]: a bit weird to check nil? but well validate-opts! would simply throw an exception if it fails

Comment (borkdude) [reply test/carve/main_test.clj]: `(is nil? ...)` always succeeds, you should write `(is (nil? ...))` or `(is (empty?  ...))`

Comment (AndreaCrotti) [reply test/carve/main_test.clj]: Ah yes of course, weirdly kaocha didn't notice that even in a test by itself, maybe it only checks the `=/not=` cases.
Anyway fixed it thanks!

**Comment by borkdude** [comment]
Is this ready for merge?

**Comment by AndreaCrotti** [comment]
If you are happy with it yes.
I'll try to use draft prs in the future when something is wip

**Pull request merged**